tests/framework/extensions/workloads/pods/pod_status.go

```diff
@@ -17,86 +16,80 @@ const (
 
 // StatusPods is a helper function that uses the steve client to list pods on a namespace for a specific cluster
 // and return the statuses in a list of strings
-func StatusPods(client *rancher.Client, clusterID string) ([]string, []error) {
+func StatusPods(client *rancher.Client, clusterID string) []error {
 	downstreamClient, err := client.Steve.ProxyDownstream(clusterID)
 	if err != nil {
-		return nil, []error{err}
+		return []error{err}
 	}
 
-	var podResults []string
 	var podErrors []error
 
 	steveClient := downstreamClient.SteveType(PodResourceSteveType)
 	err = kwait.Poll(5*time.Second, 15*time.Minute, func() (done bool, err error) {
-		podResults = []string{}
+		// emptying pod errors every time we poll so that we don't return stale errors
 		podErrors = []error{}
 
 		pods, err := steveClient.List(nil)
 		if err != nil {
-			podErrors = append(podErrors, err)
+			// not returning the error in this case, as it could cause a false positive if we start polling too early.
 			return false, nil
 		}
 
-		podResults = append(podResults, "Pod's Status: \n")
-
 		for _, pod := range pods.Data {
-			podResult, podError, err := CheckPodStatus(&pod)
-			if err != nil {
+			isReady, err := IsPodReady(&pod)
+			if !isReady {
 				// not returning the error in this case, as it could cause a false positive if we start polling too early.
 				return false, nil
 			}
-			if podError != nil {
-				podErrors = append(podErrors, podError)
-				return false, nil
-			}
-			if podResult != "" {
-				podResults = append(podResults, podResult)
+
+			if err != nil {
+				podErrors = append(podErrors, err)
 			}
 		}
-
-		if len(podResults) > 0 && len(podErrors) == 0 {
-			return true, nil
+		if len(podErrors) != 0 {
```

Comment:
If the pod isn't ready, it retriggers the poll in L42. Isn't this enforced to reset the error slice? I.e. let's say we've got a single errored pod in 3 pods, and it will be always in an error state, so therefore errors slice going to reset, so are we going to get the timeout from the poll?

I don't think we need this, cause we already handle it with isReady -> if not retrigger the poll

```
if len(podErrors) != 0 {
...
}
```

Reply:
agreed